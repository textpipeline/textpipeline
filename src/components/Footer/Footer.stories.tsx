import { storiesOf } from '@storybook/react';
import React from 'react';

import { footer } from '../../config';
import Footer from './Footer';

storiesOf('Footer', module).add('default', () => <Footer footer={footer} />);
