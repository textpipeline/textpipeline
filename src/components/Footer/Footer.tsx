import Box from '@material-ui/core/Box';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListSubheader from '@material-ui/core/ListSubheader';
import GitHubIcon from '@material-ui/icons/GitHub';
import TwitterIcon from '@material-ui/icons/Twitter';
import ListItemLink, { ListItemLinkProps } from 'components/ListItemLink';
import React from 'react';
import { ossPath } from 'routes';

interface FooterProps {
  [key: string]: ListItemLinkProps[];
}

const footer: FooterProps = {
  'Social Media': [
    { icon: TwitterIcon, text: 'Twitter', href: 'https://twitter.com/textpipeline', isExternal: true },
    { icon: GitHubIcon, text: 'GitHub', href: 'https://github.com/textpipeline', isExternal: true },
  ],
  Resources: [
    { text: 'About', href: '#', isExternal: false },
    { text: 'Open Source', href: ossPath, isExternal: false },
    { text: 'Stats', href: '#', isExternal: false },
    { text: 'FAQ', href: '#', isExternal: false },
    { text: 'Privacy Policy', href: '#', isExternal: false },
    { text: 'Terms of Use', href: '#', isExternal: false },
  ],
};

const sectionHeadings = Object.keys(footer);

const Footer: React.FC = () => (
  <footer>
    {sectionHeadings.map((sectionHeading, index) => (
      <React.Fragment key={sectionHeading}>
        <Box key={sectionHeading} mb={1}>
          <List dense component="nav" subheader={<ListSubheader>{sectionHeading}</ListSubheader>}>
            {footer[sectionHeading].map(({ icon, text, href, isExternal }) => (
              <ListItemLink
                key={`${sectionHeading}-${text}`}
                icon={icon}
                text={text}
                href={href}
                isExternal={isExternal}
              />
            ))}
          </List>
        </Box>
        {index !== sectionHeadings.length - 1 && <Divider />}
      </React.Fragment>
    ))}
  </footer>
);

export default Footer;
