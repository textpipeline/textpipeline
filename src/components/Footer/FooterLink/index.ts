export { default } from './FooterLink';
export * from './FooterLink';
