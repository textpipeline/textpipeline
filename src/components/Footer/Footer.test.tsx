import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Footer from './Footer';

describe('Footer', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<Footer />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
