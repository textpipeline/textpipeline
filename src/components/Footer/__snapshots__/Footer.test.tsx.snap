// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <Styled(MuiBox)
    key="Social Media"
    mb={1}
  >
    <WithStyles(ForwardRef(List))
      component="nav"
      dense={true}
      subheader={
        <ForwardRef(WithStyles)>
          Social Media
        </ForwardRef(WithStyles)>
      }
    >
      <ListItemLink
        href="https://twitter.com/textpipeline"
        icon={
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "TwitterIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          }
        }
        isExternal={true}
        key="Social Media-Twitter"
        text="Twitter"
      />
      <ListItemLink
        href="https://github.com/textpipeline"
        icon={
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "displayName": "GitHubIcon",
            "muiName": "SvgIcon",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          }
        }
        isExternal={true}
        key="Social Media-GitHub"
        text="GitHub"
      />
    </WithStyles(ForwardRef(List))>
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Divider)) />
  <Styled(MuiBox)
    key="Resources"
    mb={1}
  >
    <WithStyles(ForwardRef(List))
      component="nav"
      dense={true}
      subheader={
        <ForwardRef(WithStyles)>
          Resources
        </ForwardRef(WithStyles)>
      }
    >
      <ListItemLink
        href="#"
        isExternal={false}
        key="Resources-About"
        text="About"
      />
      <ListItemLink
        href="#"
        isExternal={false}
        key="Resources-Open Source"
        text="Open Source"
      />
      <ListItemLink
        href="#"
        isExternal={false}
        key="Resources-Stats"
        text="Stats"
      />
      <ListItemLink
        href="#"
        isExternal={false}
        key="Resources-FAQ"
        text="FAQ"
      />
      <ListItemLink
        href="#"
        isExternal={false}
        key="Resources-Privacy Policy"
        text="Privacy Policy"
      />
      <ListItemLink
        href="#"
        isExternal={false}
        key="Resources-Terms of Use"
        text="Terms of Use"
      />
    </WithStyles(ForwardRef(List))>
  </Styled(MuiBox)>
</footer>
`;
